subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -282,7 +282,12 @@ static int tls_ctr_drbg_random(void *ctx, unsigned char *buf, size_t len)
 {
 	ARG_UNUSED(ctx);
 
-#if defined(CONFIG_CSPRNG_ENABLED)
+#if defined(CONFIG_BUILD_WITH_TFM)
+	if (psa_generate_random(buf, len) != PSA_SUCCESS) {
+		return -EIO;
+	}
+	return 0;
+#elif defined(CONFIG_CSPRNG_ENABLED)
```

Comment:
I am not responsible for network, but I don't like all ifdef we are doing here. Zephyr has a random number generator API (and as it was discussed, it can use PSA) so lets just use it here. When we drop mbedtls API in favor of PSA we probably just use `psa_generate_random` without any ifdef.

Reply:
> I am not responsible for network, but I don't like all ifdef we are doing here. Zephyr has a random number generator API (and as it was discussed, it can use PSA) so lets just use it here. When we drop mbedtls API in favor of PSA we probably just use `psa_generate_random` without any ifdef.

Ok, so I'm going to close this PR and perhaps open a new one to fix the [side issue](https://github.com/zephyrproject-rtos/zephyr/pull/71745#issuecomment-2079773453)